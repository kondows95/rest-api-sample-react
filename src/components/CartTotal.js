import React from 'react';
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import CardActions from '@material-ui/core/CardActions'
import Box from '@material-ui/core/Box'
import Link from '@material-ui/core/Link'
import { Link as RouterLink } from 'react-router-dom'

const CartTotal = ({totalPrice}) => {
  return (
    <Box display="flex" flexDirection="column">
      <Card>
        <CardContent>
          <Box mt={1} fontSize="h6.fontSize" fontWeight={600}>
            Total {totalPrice} Ks
          </Box>
        </CardContent>
        <CardActions>
          <Button fullWidth={true} variant="contained" color="secondary">
            Confirm Order
          </Button>
        </CardActions>
        <CardContent>
          <Link to="/"
            component={RouterLink}>
            Continue Shopping
          </Link>
        </CardContent>
      </Card>
    </Box>  
  )
}

CartTotal.propTypes = {
  totalPrice: PropTypes.number.isRequired,
}

export default CartTotal
