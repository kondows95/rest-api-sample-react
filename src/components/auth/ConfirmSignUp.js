import React from 'react'
import PropTypes from 'prop-types'
import { Box, Container, Link, Button, TextField}  from '@material-ui/core'
import FormTitle from './FormTitle'

const ConfirmSignUp = React.memo(({
  authState, 
  changeAuthState,
  loading,
  error,
  email,
  confirmSignUp
}) => {
  const [form, setForm] = React.useState({email:"", password:"",confirmationCode:""})
  
  const handleChangeValue = fieldName => event => {
    const newForm = {...form}
    newForm[fieldName] =  event.target.value
    setForm(newForm)
  }
  
  const handleSubmit = event => {
    event.preventDefault()
    confirmSignUp(email, form['confirmationCode'])
  }
  
  const content = (
    <Container component="main" maxWidth="xs">
      <form onSubmit={handleSubmit}>
        <Box display="flex" flexDirection="column" mt={8}>
          <FormTitle>Confirm your email</FormTitle>
          <Box display="flex" justifyContent="center" fontWeight={600} color="error.main">
            {error}
          </Box>
          <Box width="100%" my={2}>
            <TextField
              id="confirmationCode"
              name="confirmationCode"
              label="Confirmation Code"
              onChange={handleChangeValue("confirmationCode")} 
              value={form.confirmationCode}
              variant="outlined"
              required
              fullWidth
            />
          </Box>
          <Box width="100%" mt={4} mb={2}>
            <Button
              type="submit"
              fullWidth
              variant="contained"
              color="primary"
              disabled={loading}
            >
              Confirm
            </Button>
          </Box>
          <Box width="100%" my={2}>
            <Link href="#" variant="body2" onClick={handleSubmit}>
              Resend code to {email}
            </Link>
          </Box>
        </Box>
      </form>
    </Container>
  )
  return (authState === 'confirmSignUp') ? content : null
})

ConfirmSignUp.propTypes = {
  authState: PropTypes.string.isRequired,
  changeAuthState: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
  email: PropTypes.string,
  confirmSignUp: PropTypes.func.isRequired,
}

export default ConfirmSignUp
