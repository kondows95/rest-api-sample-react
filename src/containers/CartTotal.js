import { connect } from 'react-redux'
import CartTotal from '../components/CartTotal'

export default connect(
  (state) => ({
    totalPrice: state.cart.totalPrice,
    totalQuantity:  state.cart.totalQuantity,
  })
)(CartTotal)
