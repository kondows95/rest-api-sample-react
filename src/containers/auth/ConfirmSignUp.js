import { connect } from 'react-redux'
import ConfirmSignUp from '../../components/auth/ConfirmSignUp'
import { changeAuthState, confirmSignUp } from '../../modules/auth'

export default connect(
  (state) => ({
    authState: state.auth.authState,
    loading: state.auth.loading,
    error: state.auth.error,
    email: state.auth.email,
  }),
  (dispatch) => ({
    changeAuthState: (value) => dispatch(changeAuthState(value)),
    confirmSignUp: (email, code) => dispatch(confirmSignUp(email, code)),
  })
)(ConfirmSignUp)
