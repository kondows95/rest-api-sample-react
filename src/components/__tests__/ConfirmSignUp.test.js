import React from 'react'
import ReactDOM from 'react-dom';
import { act } from 'react-dom/test-utils';
import renderer from 'react-test-renderer';
import CartTotal from '../CartTotal';
import { BrowserRouter as Router } from 'react-router-dom';
import ConfirmSignUp from '../auth/ConfirmSignUp';

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.outerHTML = '<body></body>'
  container = null
})

const valueSetter = Object.getOwnPropertyDescriptor(window.HTMLInputElement.prototype, "value").set;
let authState="confirmSignUp" ;
const changeAuthState = jest.fn(()=>{
  authState ="Signup";
});
const loading =false;
const error = "Please enter confirmation code";
const email = "a@gmail.com";
const confirmSignUp = jest.fn((email,comfirmationCode) =>{
  let cod = comfirmationCode;
});
describe("ConfirmSingup component snapshot", () => {
  it('matches the snapshot', () => {
    const ConfirmSingupSnapshot = renderer.create(
        <ConfirmSignUp 
        authState={authState}
        changeAuthState={changeAuthState}
        loading={loading}
        error={error}
        email={email}
        confirmSignUp = {confirmSignUp}
        />
    ).toJSON();
    expect(ConfirmSingupSnapshot).toMatchSnapshot();
  })
})

describe("Testing ConfirmSingup", () => {
  it('Testing textfied and confirm button', () => {
    act(() => {
      ReactDOM.render(  
      <ConfirmSignUp 
        authState={authState}
        changeAuthState={changeAuthState}
        loading={loading}
        error={error}
        email={email}
        confirmSignUp = {confirmSignUp}
        />, container);
    });
    const inputArr = document.querySelectorAll('input');
    act(() => {
      valueSetter.call(inputArr[0], '1234');
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });
    expect(inputArr[0].value).toBe('1234');
    const buttonArr = document.querySelectorAll('button');
    act(() => {
      buttonArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(confirmSignUp).toHaveBeenCalledWith("a@gmail.com","1234");
    act(() => {
      valueSetter.call(inputArr[0], '');
      inputArr[0].dispatchEvent(new Event('change', { bubbles: true }));
    });
    const linkArr = document.querySelectorAll('a');
    act(() => {
      linkArr[0].dispatchEvent(new MouseEvent('click', { bubbles: true }));
    });
    expect(confirmSignUp).toHaveBeenCalledWith("a@gmail.com","");
  })
})