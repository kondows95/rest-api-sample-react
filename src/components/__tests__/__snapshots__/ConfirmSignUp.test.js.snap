// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmSingup component snapshot matches the snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <form
    onSubmit={[Function]}
  >
    <div
      className="MuiBox-root MuiBox-root-9"
    >
      <div
        className="MuiBox-root MuiBox-root-10"
      >
        <div
          className="MuiBox-root MuiBox-root-11"
        >
          <div
            className="MuiBox-root MuiBox-root-12"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <g
                fill="none"
              >
                <path
                  d="M0 0h24v24H0V0z"
                />
                <path
                  d="M0 0h24v24H0V0z"
                  opacity=".87"
                />
              </g>
              <path
                d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
              />
            </svg>
          </div>
        </div>
        <div
          className="MuiBox-root MuiBox-root-22"
        >
          Confirm your email
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-23"
      >
        Please enter confirmation code
      </div>
      <div
        className="MuiBox-root MuiBox-root-24"
      >
        <div
          className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <label
            className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined Mui-required Mui-required"
            data-shrink={false}
            htmlFor="confirmationCode"
          >
            Confirmation Code
            <span
              className="MuiFormLabel-asterisk MuiInputLabel-asterisk"
            >
              â€‰
              *
            </span>
          </label>
          <div
            className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
            onClick={[Function]}
          >
            <fieldset
              aria-hidden={true}
              className="PrivateNotchedOutline-root-82 MuiOutlinedInput-notchedOutline"
              style={
                Object {
                  "paddingLeft": 8,
                }
              }
            >
              <legend
                className="PrivateNotchedOutline-legend-83"
                style={
                  Object {
                    "width": 0.01,
                  }
                }
              >
                <span
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "&#8203;",
                    }
                  }
                />
              </legend>
            </fieldset>
            <input
              aria-invalid={false}
              className="MuiInputBase-input MuiOutlinedInput-input"
              disabled={false}
              id="confirmationCode"
              name="confirmationCode"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              required={true}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="MuiBox-root MuiBox-root-84"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-fullWidth"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Confirm
          </span>
        </button>
      </div>
      <div
        className="MuiBox-root MuiBox-root-105"
      >
        <a
          className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-body2 MuiTypography-colorPrimary"
          href="#"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
        >
          Resend code to 
          a@gmail.com
        </a>
      </div>
    </div>
  </form>
</main>
`;
